core-java/src/main/java/org/spine3/server/RepositoryBase.java

```diff
@@ -47,16 +49,29 @@ protected RepositoryBase() {
     }
 
     private Constructor<E> getEntityConstructor() {
-        final Constructor<E> result;
+        final Class<E> entityClass = getEntityClass();
+        final Class<I> idClass = getIdClass();
         try {
-            final Class<E> entityClass = getEntityClass();
-            final Class<I> idClass = getIdClass();
+            final Constructor<E> result = entityClass.getDeclaredConstructor(idClass);
+            checkConstructorAccessModifier(result.getModifiers(), entityClass.getSimpleName());
+            return result;
+        } catch (NoSuchMethodException ignored) {
+            throw noSuchConstructorException(entityClass.getSimpleName());
```

Comment:
Having short class name may be confusing. There can be cases of large applications that have the same name of things, but in different bounded contexts. E.g. consider "Order" for in two parts of a larger app that deal with different kinds of orders.

Please pass FQN.